src/components/Proposer/Proposer.svelte

```diff
@@ -0,0 +1,88 @@
+<script lang="ts">
+	import PeopleIcon from '$components/icons/PeopleIcon.svelte';
+	import PoliticianIcon from '$components/icons/PoliticianIcon.svelte';
+	import type { Assembly } from '$models/assembly';
+	import type { Party } from '$models/party';
+	import type { Politician } from '$models/politician';
+	import dayjs from 'dayjs';
+	import 'dayjs/locale/th';
+	import buddhistEra from 'dayjs/plugin/buddhistEra';
+
+	dayjs.extend(buddhistEra);
+	dayjs.locale('th');
+
+	type CommonProposer = {
+		name: string;
+		description?: string;
+	};
+
+	type PartyPolitician = {
+		politician: Politician;
+		assembly: Assembly;
+		party: Party;
+	};
+
+	export let partyPolitician: PartyPolitician | undefined = undefined;
+	export let assembly: Assembly | undefined = undefined;
+	export let common: CommonProposer | undefined = undefined;
+
+	$: proposerName = () => {
+		if (partyPolitician) {
+			const { politician } = partyPolitician;
+			return [politician.prefix ?? '', politician.firstname, politician.lastname].join(' ');
+		} else if (assembly) {
+			return assembly.name;
+		} else if (common) {
+			return common.name;
+		}
+
+		return '';
+	};
+
+	const getAssemblyTermText = (assembly: Assembly) =>
+		`ชุดที่ ${assembly.term} (${dayjs(assembly.startedAt).format('BBBB')})`;
+
+	$: proposerTerm = () => {
```

Comment:
Same feedback with `proposerName`

But since the function is not complicated, ternary operator `condition ? true : false` also works. (Either way is fine for me)